File: leetcode/Q104_MaximumDepthofBinaryTree.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if root is None:
            return 0
        if root.left is None and root.right is None:
            return 1
        else:
            max_num = 0
            if root.left is not None:
                max_num = max(self.maxDepth(root.left) + 1, max_num)
            if root.right is not None:
                max_num = max(self.maxDepth(root.right) + 1, max_num)

            return max_num

File: src/easy/Q104_MaximumDepthofBinaryTree.java
package easy;

public class Q104_MaximumDepthofBinaryTree {

    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
          this.val = val;
          this.left = left;
          this.right = right;
        }
    }
    class Solution {
        public int maxDepth(TreeNode root) {
            return findDepth(root, 0);
        }

        public int findDepth(TreeNode node, int depth){
            if(node == null) return depth;
            return Math.max(findDepth(node.left, depth + 1), findDepth(node.right, depth + 1));
        }
    }
}
